compiler/luci/profile/src/CircleNodeOrigin.cpp

```diff
@@ -144,13 +144,14 @@ bool has_origin(const luci::CircleNode *circle_node)
   return circle_node->annot<CircleNodeOriginAnnotation>() != nullptr;
 }
 
+/**
+ * @brief 'origin' is added to the existing origin of circle_node.
+ */
 void add_origin(luci::CircleNode *circle_node, const std::shared_ptr<CircleNodeOrigin> origin)
 {
-  if (origin == nullptr)
-    return;
-
+  auto new_origin = composite_origin({get_origin(circle_node), origin});
```

Comment:
(optional) it would be nice to have an `assert(origin != nullptr)`

Reply:
`origin` can be `nullptr` for now :)

```cpp
add_origin(circle_node, nullptr) = get_origin(circle_node)
add_origin(circle_node, originA) = get_origin(circle_node) + originA
```

This is done by following code. `nullptr` is ignored :)
https://github.com/Samsung/ONE/blob/cf3f7ccb945feb43658c8b5c4f15b77a2684feb3/compiler/luci/profile/src/CircleNodeOrigin.cpp#L83-L93

Reply:
> origin can be nullptr for now :)

Um... from the code in this function (not looking into `CompositeOrigin`) this seems implicit feature.
Could you add an comment that `origin` can be nullptr and it there will be no changes then?